src/Features/Core/Portable/Shared/Options/ServiceFeatureOnOffOptions.cs

```diff
@@ -35,5 +35,34 @@ public static bool IsClosedFileDiagnosticsEnabled(OptionSet options, string lang
 
             return option.Value;
         }
+
+        /// <summary>
+        /// Option to disable analyzer execution during live analysis.
+        /// </summary>
+        public static readonly Option<bool> DisableAnalyzers = new Option<bool>(
+            nameof(ServiceFeatureOnOffOptions), nameof(DisableAnalyzers), defaultValue: false,
+            storageLocations: new RoamingProfileStorageLocation($"Options.DisableAnalyzers"));
+
+        /// <summary>
+        /// Option to turn off all background analysis to improve performance.
+        /// </summary>
+        public static readonly Option<bool> PowerSaveMode = new Option<bool>(
+            nameof(ServiceFeatureOnOffOptions), nameof(PowerSaveMode), defaultValue: false,
+            storageLocations: new RoamingProfileStorageLocation($"Options.PowerSaveMode"));
+
+        /// <summary>
+        /// Enables forced power save mode when low VM is detected to improve performance.
```

Comment:
❕ Power save mode is not designed to improve performance and not designed to reduce memory consumption. Low VM is not a reasonable automatic trigger for this feature. If we want an automatic trigger, it should be tied to the operating system's current power mode.

Note that I already have a hard-coded exception in the low VM monitor to exclude it from running on my machine, but we really should not be making this change for other people either.

Reply:
I am going to start a separate offline email thread on this - I think we need two modes here:

1. Power save mode: Only about battery/CPU consumption, without any aim at reducing memory consumption or turning off potentially expensive features. This can be tied to OS power mode if required.
2. Lightweight edit mode or resource saver mode: Turn off any background analysis that can lead to high CPU consumption or help reduce memory consumption. It is also reasonable to turn off features in this mode if they are not an absolute requirement for minimal IDE editing experience. Given the memory save characteristics of this mode, it is also likely that some explicitly triggered IDE features might perform worse performance-wise, we would look at these case by case basis and tune the things that get turned off. This mode can be tied to OS power mode and/or low VM trigger.

My PR is basically targeting the later, and based on my discussions with @heejaechang @jinujoseph and @vatsalyaagrawal later is what the customers have really been requesting. Each of the above modes serve different purposes and we should meet up to decide the priority for these.